Feature/publications biblatex

aminmkhan opened the pull request:
Added support for including publications through BibLaTeX, based on Fancy CV template created by Adrien Friggeri.

chozelinek commented:
This feature would be awesome!

johannesbottcher commented:
The bibliography part of the friggeri cv template broke with the last update of `biblatex`, threw lots of warnings before, and does some real strange stuff. I would advice against merging that in its current state.

aminmkhan commented:
Fixed the issue caused by changes in **`\DeclareNameFormat`** in `biblatex` (see plk/biblatex#372). This should be working fine now.

thiagoveloso commented:
Nice addition, folks! Any idea on how to match the Publications font to the rest of the document?

aminmkhan commented:
@thiagoveloso You can use `\setmainfont`.

See https://github.com/aminmkhan/Awesome-CV/blob/6c08ff11ff1c388460d60046a1ad0c8b96b2465e/awesome-cv.cls#L202-L209

thiagoveloso commented:
@aminmkhan, thank you very much. Haven't used the `\setmainfont` command, but the changes merged from @jckegelman made my CV's "Publications" section look much better. Help much appreciated.

andrewcassidy commented:
Any way to get hyperlinks to the urls?

aminmkhan commented:
@andrewcassidy One quick workaround could be to use the `note` field in `*.bib` entries. 

Better solution would be to play around with `\DeclareFieldFormat` in the class file `awesome-cv.cls`.

SiggiSmara commented:
I wanted more of the Awesome look for the bibtex entries and took a swipe at achieving that. See here for the end result:
![selection_018](https://user-images.githubusercontent.com/8719966/39358633-ff83f0c8-4a16-11e8-9011-b51125192999.png)
I don't want to tread on anyone's toes but I'm happy to share my changes. If anyone is interested then let me know and I'll see if I can figure out how to add my changes to this pull request.  
Awesome job everybody!

aminmkhan commented:
Thanks @SiggiSmara. For now you can push your changes to a pull request similar to aminmkhan/Awesome-CV#2, and we'll see.

SiggiSmara commented:
Cool @aminmkhan . Pushed the changes to you. First time for me to push from one fork to another so let me know if I managed to make a mistake along the way.

omrysendik commented:
@SiggiSmara can you kindly share an example of how you create a publication entry?
Am I missing it inside @aminmkhan repository? Couldn't find it.

SiggiSmara commented:
Sorry @omrysendik  I missed your comment. Are you still having trouble?  In the examples/cv folder you should see a bibtex file (references.bib) and a publications.tex file.  The bibtex file contains the reference information itself while the publications.tex file imports the references that you want to appear.
Let me know if I can help...

aminmkhan commented:
@omrysendik You can check the **Usage** section in `README` [here](https://github.com/aminmkhan/Awesome-CV/tree/feature/publications-biblatex#list-of-publications) or [in PR](https://github.com/posquit0/Awesome-CV/pull/23/files?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8).

syoh commented:
Thanks for these commits. I made my publication section work based on the latest commit in this thread.

![download](https://user-images.githubusercontent.com/1441512/44321074-c100b280-a3fa-11e8-8f18-fbcff24d51dd.png)

There is one thing I cannot figure out is how to use first initials instead of full first names. Based on some search, it seems I have to use `giveninits=true` somewhere. My best guess are these mods to `awesome-cv.cls`:

```
\RequirePackage[ % BibLaTeX
sorting=ydnt, % Sorts entries by year (descending order), name, title
style=verbose,
doi=false,
isbn=true,
url=false,
eprint=false,
backref = false, % include back references in bibliography
maxcitenames=3, % affects only the citations in the document body
maxbibnames=99, % affects only the bibliography, pass 99 to print all
hyperref=true,
block=none,
uniquename=init,
giveninits=true,
backend=biber % {Options: bibtex, biber}
]{biblatex}
```

However, the changes have completely no effect (I am using sharelatex.com). Additionally, I tried putting the options in different places; however, my publications section seems to ignore it.

How I can use first initials instead of full first names? 

I would appreciate any help

jckegelman commented:
@syoh the `giveninits=true` option is ignored because `awesome-cv.cls` has its own `\DeclareNameFormat` that uses `\namepartgiven`. If you always want to use initials, you can replace `\namepartgiven` with `\namepartgiveni`. If you want it to honor the `giveninits` option, you can use `\ifgiveninits{\namepartgiveni}{\namepartgiven}`

ashwinvis commented:
@syoh Your changes look really elegant. Can you commit and push them online?

ShanSabri commented:
Is there a way to format an author in bold? Also a way to number publications?

aminmkhan commented:
@ShanSabri You can try numerical, though there could be [some issues](https://github.com/aminmkhan/Awesome-CV/issues/5).

You'll have to play with [custom code](https://github.com/aminmkhan/Awesome-CV/blob/feature/publications/awesome-cv.cls#L732-L836) to change formatting for the author.

ShanSabri commented:
@syoh How did you get your publication title to be bold?

ashwinvis commented:
@ShanSabri aminmkhan/Awesome-CV#6

aminmkhan commented:
Thanks @ashwinvis, I will try to integrate this and update the #23.

ShanSabri commented:
@aminmkhan Unfortunately I am running into some issues trying to numerate publications. Any additional advice or working example would be greatly appreciated.

aminmkhan commented:
@ShanSabri You can try having a look at some [recent pull requests](https://github.com/aminmkhan/Awesome-CV/pulls), I haven't used numbered list in this style myself.

jamesadevine commented:
Thanks @aminmkhan for contributing such a useful feature to the template.

However, if using this template for research statements where numerical citations are required references may appear like this:

![Screenshot 2019-11-20 at 18 03 15](https://user-images.githubusercontent.com/4742697/69264847-08f32400-0bc0-11ea-867c-c5025612df23.png)

To fix the period between entries, define the following:

```
\renewcommand*\newunitpunct{}
```

It's up to you whether you want to integrate this into the PR @aminmkhan :smile:

aminmkhan commented:
Thanks @jamesadevine, also mentioned in https://github.com/aminmkhan/Awesome-CV/issues/5.

hyyoonzvolti commented:
Hi, Thanks for the great template. 

I was wondering if there is any way to underlining or bold-texting one of the author of the paper. It is more clear to indicate the applicant's contribution to the paper when presented to the highering manager.

inmybrain commented:
@hyyoonzvolti 

Hi, would you like to see [this issue](https://github.com/posquit0/Awesome-CV/issues/292#issue-480130000) that I answered by myself?

hyyoonzvolti commented:
> @hyyoonzvolti
> 
> Hi, would you like to see [this issue](https://github.com/posquit0/Awesome-CV/issues/292#issue-480130000) that I answered by myself?

Hi, Thanks for the suggestion.

As stated in the [link](https://tex.stackexchange.com/questions/73136/make-specific-author-bold-using-biblatex/327046?newreg=64f7c699ce174c84addfe59640a16cce
), the none of the solutions provided here work for biblatex > 3.11. I am using "3.14", so it does not for my case.


The [alternative](https://tex.stackexchange.com/questions/187181/independent-publication-list-with-numbered-list-using-biblatex-and-refsection
) to use function **forcsvlist** does NOT work too.

`\forcsvlist{\listadd\boldnames}
  {{Pospiech, Matthias}, {Pospiech, M.}}
`

CBornes commented:
Hi
I was wondering if there is a way to show the numbers in reverse order.
[3] Most recent
[2] Middle
[1] Older

ashwinvis commented:
@CBornes always [search online](https://tex.stackexchange.com/questions/21401/biblatex-reverse-numbering-i-e-count-down) first :smile:

CBornes commented:
> @CBornes always [search online](https://tex.stackexchange.com/questions/21401/biblatex-reverse-numbering-i-e-count-down) first 😄

Ups sorry I've found some answers (not this) and all of them broke the formatting. But this one works like a charm thanks.

lmontigny commented:
So how to add the list of publication from a bib file?

Thanks

navidlilian commented:
> I wanted more of the Awesome look for the bibtex entries and took a swipe at achieving that. See here for the end result:
> ![selection_018](https://user-images.githubusercontent.com/8719966/39358633-ff83f0c8-4a16-11e8-9011-b51125192999.png)
> I don't want to tread on anyone's toes but I'm happy to share my changes. If anyone is interested then let me know and I'll see if I can figure out how to add my changes to this pull request.
> Awesome job everybody!
@ SiggiSmara, I am pretty new to Latex, I used the new files which included the publications but none of them work for me. Not sure what is missed. Is it possible that you could share the code package you used to make your publications so I can use it as my resume tempelate? Many thanks in advance.

donadviser commented:
> Hi
> I was wondering if there is a way to show the numbers in reverse order.
> [3] Most recent
> [2] Middle
> [1] Older

I have solved this publication problem. See the attached.
![SmartSelect_20210816-080248_Box](https://user-images.githubusercontent.com/31127659/129524291-70bb8624-09b1-42db-ab1f-56bab81cebcd.jpg)

LuPo commented:
Hi @donadviser, you can use "reversenumbering=true" in the biblatex package options

navidlilian commented:
Hello everyone, I am pretty new to Latex,  I am trying to add the publications to my resume but cannot figure how and the publication section appears as empty! I would appreciate it if anyone can help me with this issue.
![image](https://user-images.githubusercontent.com/88998642/129848011-89325f2e-2018-46af-afb2-d40c7b4e10f0.png)

aminmkhan commented:
> I wanted more of the Awesome look for the bibtex entries and took a swipe at achieving that. See here for the end result:

Thanks @navidlilian, you can push your code changes as pull request here: https://github.com/aminmkhan/Awesome-CV/pulls

The merge and clean-up is long overdue, but do push your changes and we can see how best to integrate. Thank you!


> Is it possible that you could share the code package you used to make your publications so I can use it as my resume template? Many thanks in advance.

To include publications, you need to include them as `*.bib` file, like the example here: https://github.com/aminmkhan/Awesome-CV/blob/feature/publications-biblatex/examples/cv/references.bib

If you are unfamiliar, check [BibLaTeX](https://www.overleaf.com/learn/latex/Articles/Getting_started_with_BibLaTeX) and there are tools like [JabRef](https://www.jabref.org/) that help creating and managing bibliography easier!

donadviser commented:
> Hi @donadviser, you can use "reversenumbering=true" in the biblatex package options

You are right. That's what I did. I'm OK what how mine looked. Thanks

aminmkhan commented on README.md:
````diff
@@ -57,6 +57,19 @@ $ xelatex {your-cv}.tex
+```bash
+$ xelatex {your-cv}.tex
+$ biber {your-cv}
+$ xelatex {your-cv}.tex
````
Let me test again to see if this needs to be updated.

aminmkhan replied:
Yes, we need to run `xelatex` once before `biber`.

For example, [see this example here](https://en.wikibooks.org/wiki/LaTeX/Bibliographies_with_biblatex_and_biber#A_simple_example).

aminmkhan commented on awesome-cv.cls:
```diff
@@ -670,3 +670,151 @@
+   \widowpenalty10000
+   \interlinepenalty5000}
+
+% Customized format, based on Fancy CV template created by Adrien Friggeri
```
As seen here, searching for GitHub source, also available at [Overleaf](https://www.overleaf.com/latex/templates/friggeri-cv-template/hmnchbfmjgqh) ...

> Original by Adrien Friggeri (MIT license)
> with modifications by Alejandro Pérez Londoño (CC-BY)

aminmkhan replied:
Included license and the link to original repo.

avalada commented:
> > Hi
> > I was wondering if there is a way to show the numbers in reverse order.
> > [3] Most recent
> > [2] Middle
> > [1] Older
> 
> I have solved this publication problem. See the attached. ![SmartSelect_20210816-080248_Box](https://user-images.githubusercontent.com/31127659/129524291-70bb8624-09b1-42db-ab1f-56bab81cebcd.jpg)

How did you define this style?

donadviser commented:
> Hi @donadviser, you can use "reversenumbering=true" in the biblatex package options

Have you tried this method? Let me know if it didn't solve it, then the next option is to use calculating the total number of entries in each reference section.

avalada commented:
> > Hi @donadviser, you can use "reversenumbering=true" in the biblatex package options
> 
> Have you tried this method? Let me know if it didn't solve it, then the next option is to use calculating the total number of entries in each reference section.

Sorry I meant the entire style with the numbering, author, "title", journal (italics). etc.

Currently, I have this`\DeclareBibliographyDriver{article}{%
  \setlength{\parskip}{0pt}
  \fontsize{9pt}{1em}\bodyfont\color{black}%
  {\bullet\hspace{1.5ex}}\printfield{title}%
  \newblock%
  \printnames{author}%
  \par%
  \newblock%
  {%
    \fontsize{8pt}{1em}\bodyfont\scshape\color{graytext}%
    \usebibmacro{journal+issuetitle}%
    \setunit{\space}%
    \printfield{pages}%
    \newunit%
    \printlist{publisher}%
    \setunit*{\addcomma\space}%
    \printfield{year}%
    \setunit*{\addcomma\space}%
    \color{awesome}\printfield{note}%
    \newunit%
  }
  \par\vspace{0.8\baselineskip}
}`

donadviser commented:
> > > Hi @donadviser, you can use "reversenumbering=true" in the biblatex package options
> > 
> > 
> > Have you tried this method? Let me know if it didn't solve it, then the next option is to use calculating the total number of entries in each reference section.
> 
> Sorry I meant the entire style with the numbering, author, "title", journal (italics). etc.
> 
> Currently, I have this`\DeclareBibliographyDriver{article}{% \setlength{\parskip}{0pt} \fontsize{9pt}{1em}\bodyfont\color{black}% {\bullet\hspace{1.5ex}}\printfield{title}% \newblock% \printnames{author}% \par% \newblock% {% \fontsize{8pt}{1em}\bodyfont\scshape\color{graytext}% \usebibmacro{journal+issuetitle}% \setunit{\space}% \printfield{pages}% \newunit% \printlist{publisher}% \setunit*{\addcomma\space}% \printfield{year}% \setunit*{\addcomma\space}% \color{awesome}\printfield{note}% \newunit% } \par\vspace{0.8\baselineskip} }`

For my article, I used the following
\DeclareBibliographyDriver{article}{%
\fontsize{10pt}{1.0em}\bodyfontlight\upshape\color{text}
  \usebibmacro{bibindex}%
 \newunit
  \usebibmacro{begentry}%
  \usebibmacro{author/translator+others}%
  \setunit{\labelnamepunct}\newblock
  \usebibmacro{title}%
  \newunit
  \newunit\newblock
  \usebibmacro{byauthor}%
  \newunit\newblock
  \usebibmacro{bytranslator+others}%
  \newunit\newblock
  \printfield{version}%
  \newunit\newblock%
  \usebibmacro{journal+issuetitle}%
  \newunit
  \newunit\newblock
  \usebibmacro{byeditor+others}%
  \newunit
  \usebibmacro{note+pages}%
  \newunit\newblock
  \usebibmacro{doi+eprint+url}%
  \newunit\newblock
  \usebibmacro{addendum+pubstate}%
  \setunit{\bibpagerefpunct}\newblock
  \usebibmacro{pageref}%
  \usebibmacro{finentry}
  \par\vspace{0.5\baselineskip}}

avalada commented:
@donadviser Doesnt give me the same output as your screenshot. I dont get numbers and the journal appears in the new line. Would you mind sharing your cls file?
Thanks for your help!

donadviser commented:
@avalada, can you show what it looks like?

avalada commented:
@donadviser Also attached my cls file
<img width="171" alt="Screenshot 2022-01-27 at 8 51 45 PM" src="ht=ps://
[awesome-cv.txt](https://github.com/posquit0/Awesome-CV/files/7953412/awesome-cv.txt)
user-images.githubusercontent.com/4357052/151433616-091d5288-bd37-4757-af95-f391248a2adf.png">

donadviser commented:
@avalada Let me know if the below will fix your publications

%-------------------------------------------------------------------------------
%                Bibliography
%-------------------------------------------------------------------------------
\RequirePackage[ % BibLaTeX
	sorting=ydnt, % Sorts entries by year (descending order), name, title
	bibstyle=numeric,
	doi=false,
	isbn=true,
	url=false,
	eprint=false,
	backref = false, % include back references in bibliography
	maxcitenames=3, % affects only the citations in the document body
	maxbibnames=99, % affects only the bibliography, pass 99 to print all
	hyperref=true,
	block=none,
	backend=biber % {Options: bibtex, biber}
	]{biblatex}
%	
%--------------------------------------------------------------
% Count total number of entries in each refsection
\AtDataInput{%
  \csnumgdef{entrycount:\therefsection}{%
    \csuse{entrycount:\therefsection}+1}
}

% Print the labelnumber as the total number of entries in the
% current refsection, minus the actual labelnumber, plus one
\DeclareFieldFormat{labelnumber}{\mkbibdesc{#1}}
\newrobustcmd*{\mkbibdesc}[1]{%
  \number\numexpr\csuse{entrycount:\therefsection}+1-#1\relax}
%-------------------------------------------------------------

% Avoid splitting entries across page break (only for 3 or fewer lines)
% Tip: http://tex.stackexchange.com/a/51261
\AtBeginEnvironment{thebibliography}{%
   \clubpenalty10000
   \@clubpenalty \clubpenalty
   \widowpenalty10000
   \interlinepenalty5000}

\DeclareBibliographyDriver{article}{%
\fontsize{10pt}{1.0em}\bodyfontlight\upshape\color{text}
  \usebibmacro{bibindex}%
  \newunit
  \usebibmacro{begentry}%
  \usebibmacro{author/translator+others}%
  \setunit{\labelnamepunct}\newblock
  \usebibmacro{title}%
  \newunit
  \newunit\newblock
  \usebibmacro{byauthor}%
  \newunit\newblock
  \usebibmacro{bytranslator+others}%
  \newunit\newblock
  \printfield{version}%
  \newunit\newblock%
  \usebibmacro{journal+issuetitle}%
  \newunit
  \newunit\newblock
  \usebibmacro{byeditor+others}%
  \newunit
  \usebibmacro{note+pages}%
  \newunit\newblock
  \usebibmacro{doi+eprint+url}%
  \newunit\newblock
  \usebibmacro{addendum+pubstate}%
  \setunit{\bibpagerefpunct}\newblock
  \usebibmacro{pageref}%
  \usebibmacro{finentry}
  \par\vspace{0.5\baselineskip}
  }

avalada commented:
@donadviser almost! Some of the entries have an indentation as in the sceeenshot, any idea how to remove this ident?
<img width="149" alt="Screenshot 2022-01-30 at 6 50 09 PM" src="https://user-images.githubusercontent.com/4357052/151711094-ab1fa54a-f132-4e0e-8720-9ba5d94344f2.png">

OJFord commented:
Closes #271

aminmkhan commented:
> publication with numbering

@avalada and @donadviser  That's great to see it working, if you want you can submit your changes as pull request either here, or at: https://github.com/aminmkhan/Awesome-CV/pulls

Thank you.

aminmkhan commented on examples/cv.tex:
```diff
@@ -93,7 +97,8 @@
 \input{cv/honors.tex}
 \input{cv/presentation.tex}
-\input{cv/writing.tex}
+%\input{cv/writing.tex}
```
I am going to undo this comment out, and will update in the pull request.

Still thinking of the best way to include publications sections in the original CV, without conflicting with the existing _Writing_ section included by Byungjin.

aminmkhan replied:
I have left the `writing.tex` unchanged, and added comments on how to include new publications section.

aminmkhan commented on awesome-cv.cls:
```diff
@@ -670,3 +670,151 @@
+    \printfield{year}%
+    \setunit{\addcomma\space}%
+    \printlist{location}%
+    \newunit%
```
Thanks, moved this discussion here: https://github.com/aminmkhan/Awesome-CV/issues/11

aminmkhan commented on awesome-cv.cls:
```diff
@@ -670,3 +670,151 @@
+\DeclareFieldFormat[misc]{title}{#1\par}
+\DeclareFieldFormat[report]{title}{#1\par}
+\DeclareFieldFormat[incollection]{title}{#1\par}
+
```
Moved as https://github.com/aminmkhan/Awesome-CV/issues/12

aminmkhan commented on awesome-cv.cls:
```diff
@@ -670,3 +670,151 @@
+  \par\vspace{0.3\baselineskip}
+}
+
+\DeclareBibliographyDriver{inproceedings}{%
```
Moved as https://github.com/aminmkhan/Awesome-CV/issues/12

OJFord commented:
Please do submit here, or update yours if merged in your repo @aminmkhan. This is a very long-standing PR but seems like an excellent addition, I do mean to review properly with a view to merging soon. Thanks.

aminmkhan commented:
Thank a lot @OJFord. I have now updated this pull request, with fresh rebase from `master` and squashing all commits.

_Ready to review and land this!_ 🚀 

I have moved some of the suggestions and conversations as issues to aminmkhan#11, aminmkhan#12, aminmkhan#13, aminmkhan#14, and will update in future pull requests.

aminmkhan commented:
Just checking @OJFord, can we proceed with the review and merging this PR? Thanks!

sjiang95 commented:
Awesome work! Looking forward to being merged!

OJFord commented on examples/cv.tex:
```diff
@@ -105,6 +110,8 @@
 \input{cv/honors.tex}
 \input{cv/presentation.tex}
 \input{cv/writing.tex}
+% For publications from bibtex, add to cv/references.bib and uncomment below
+% \input{cv/publications.tex}
```
I think it'd be good to have the example complete - so there's an example of rendered result, and an example of what references.bib should actually look like.

aminmkhan replied:
Yes @OJFord I wanted to update the example, but couldn't decide since it doesn't make much sense to update the main example included in [preview](https://github.com/posquit0/Awesome-CV#preview)

Also putting in fictitious bibliography entries to the awesome original creator's CV included in preview wouldn't look nice. 

As an alternative, should we upload another example? e.g.

`https://github.com/posquit0/Awesome-CV/blob/master/examples/resume-with-publications.pdf`

OJFord commented:
Apart from that I just want to test a couple of CVs myself and I think good to merge. 👍 Thanks!

sjiang95 commented on README.md:
````diff
@@ -99,6 +99,23 @@ $ docker run --rm --user $(id -u):$(id -g) -i -w "/doc" -v "$PWD":/doc thomaswei
+
+```bash
+$ xelatex {your-cv}.tex
+$ biber {your-cv}
````
I have a small question about biber: what is the motivation to use biber instead of bibtex?
When I test this commit today (20230729), I found that the (latest after executing `tlmgr update --self --all`) texlive 2023 installed biber package but there is no biber binary, for which I had some troubles with this awesome feature.
Is it possible to use bibtex instead of biber cause the former is a common choice to handle bibliography?

aminmkhan replied:
@sjiang95 I think you can fairly easily replace the commands with `bibtex` locally if you are comfortable working with LaTeX. See package settings [here](https://github.com/aminmkhan/Awesome-CV/blob/67c15a831b5b776d5d0eb5739dfed0c7889b1e9b/awesome-cv.cls#L812).

```
backend=bibtex
```


`biber` generally has better features, you can try reading more [here](https://www.overleaf.com/learn/latex/Bibliography_management_with_biblatex) and [here](https://tex.stackexchange.com/a/25702) in case you are curious.

sjiang95 replied:
Thanks for your kind explanation. You've done an awesome job.

sjiang95 commented:
I have tested this pr and believe that it is ready to be merged.👍

mbhall88 commented:
Any news on merging this @OJFord?

JJGabe commented:
I'm not able to get the bibliography section to work for my CV. I copied the entire Bibliography section from the awesome-cv.cls of @aminmkhan into my awesome-cv.cls (I have made some other formatting edits for my own use) and then added a references.tex file, following the example, and put `\addbibresource{references.bib}` in the preamble of my cv.tex file. However, I am left with just the section and subsection headings; the references are not filled in (see screenshot).
![BlankPubs](https://github.com/posquit0/Awesome-CV/assets/64604797/7e1f37ab-6bb2-4b9b-b910-a88933c8b564)

I'm using TeXStudio on Ubuntu 22.04 and have a full and up-to-date texlive installation.

aminmkhan commented:
@JJGabe you also need to include the `references.bib` file with some entries in BibTeX format. The [examples/cv/references.bib](https://github.com/posquit0/Awesome-CV/blob/67c15a831b5b776d5d0eb5739dfed0c7889b1e9b/examples/cv/references.bib#L3) included is empty.

JJGabe commented:
@aminmkhan Sorry, I should have been clearer. I did include my own BibTex file, which I exported from Zotero as a BetterBibTex, but I renamed it references.bib to match your template. Even when I have compiled the CV once, with the `\addbibresource` included, the autofill options don't show up when I start a `nocite()` command in the `references.tex` file, as they usually do with the other articles I have written.
![Sample-bibtex](https://github.com/posquit0/Awesome-CV/assets/64604797/3d592528-b7e9-4fd6-8060-98972fb46095)

![Sample-references](https://github.com/posquit0/Awesome-CV/assets/64604797/192dfa78-b924-462a-bed6-515657906f13)

aminmkhan commented:
@JJGabe Not sure what's happening, perhaps cross-check that the path to bib file is correct, or check the output log of `make` / `xelatex` command for any errors.

```
\addbibresource{cv/references.bib}
```

If still not resolved, perhaps report an issue at [aminmkhan/Awesome-CV/issues](https://github.com/aminmkhan/Awesome-CV/issues/) to follow-up. Thanks.

JJGabe commented:
@aminmkhan, thanks for the response. I made a couple of changes, including adding an extra Biber compile in my build process (xelatex -> Biber -> Biber -> xelatex) and moving the `\addbibresource{}` call to the `awesome-cv.cls` file, immediately following the `\RequirePackage{biblatex}` block. My references are now working, though I'm not sure which change was responsible....maybe a combination of both.
